core/simulations/steps/createVaultForCollateral.ts

```diff
@@ -0,0 +1,160 @@
+import { setCollateralInVat } from '../../helpers/hardhat';
+import { getCollateralConfigByType } from '../../src/constants/COLLATERALS';
+import BigNumber from '../../src/bignumber';
+import { changeVaultContents, fetchVault, openVault, fetchVaultCollateralParameters } from '../../src/vaults';
+import { HARDHAT_PUBLIC_KEY, TEST_NETWORK } from '../../helpers/constants';
+import { RAD_NUMBER_OF_DIGITS, WAD_NUMBER_OF_DIGITS } from '../../src/constants/UNITS';
+import getContract, {
+    getContractAddressByName,
+    getErc20Contract,
+    getJoinNameByCollateralType,
+} from '../../src/contracts';
+import { depositCollateralToVat, fetchCollateralInVat, withdrawCollateralFromVat } from '../../src/wallet';
+import { MAX } from '../../src/constants/UNITS';
+import { CollateralType } from '../../src/types';
+import { ethers } from 'ethers';
+
+const getLatestVault = async () => {
+    const cdpManager = await getContract(TEST_NETWORK, 'CDP_MANAGER', true);
+    const lastHex = await cdpManager.last(HARDHAT_PUBLIC_KEY);
+    return new BigNumber(lastHex._hex).toNumber();
+};
+
+const setupCollateralInVat = async (collateralType: CollateralType, collateralOwned: BigNumber) => {
+    console.info('Setting collateral in VAT...');
+    await setCollateralInVat(collateralType, collateralOwned);
+    const balance = await fetchCollateralInVat(TEST_NETWORK, HARDHAT_PUBLIC_KEY, collateralType);
+    if (!balance.eq(collateralOwned)) {
+        throw new Error(
+            `Unexpected vat balance. Expected: ${collateralOwned.toFixed()}, Actual: ${balance.toFixed()}`
+        );
+    }
+    console.info(`Vat Collateral ${collateralType} balance of ${HARDHAT_PUBLIC_KEY} is ${collateralOwned.toFixed()}`);
+    return balance;
+};
+const giveMaxAllowance = async (addressJoin: string, tokenContractAddress: string) => {
+    const contract = await getErc20Contract(TEST_NETWORK, tokenContractAddress, true);
+    await contract.approve(addressJoin, MAX.toFixed(0));
+    console.info('Max allowance given out');
+};
+const extractCollateralFromVat = async (
+    collateralType: CollateralType,
+    collateralOwned: BigNumber,
+    tokenContractAddress: string
+) => {
+    console.info('Extracting collateral');
+    const addressJoin = await getContractAddressByName(TEST_NETWORK, getJoinNameByCollateralType(collateralType));
+    const joinContractHasSufficientFunds = await isBalanceGreaterThan(
+        TEST_NETWORK,
+        tokenContractAddress,
+        addressJoin,
+        collateralOwned
+    );
+    if (!joinContractHasSufficientFunds) {
+        throw new Error('Join contract does not have sufficient funds');
+    }
```

Comment:
Can you please explain why is this error again? Can't we just add more tokens to the join contract? Or this will effectively mean that we need to modify every token contract, which we tried to avoid?

Reply:
>  Can't we just add more tokens to the join contract? 

yes, this will effectively mean that we need to modify every token contract so it would possess enough funds to send to the user's wallet.

Reply:
That's actually a problem for testing newly onboarded collateral which have no vaults. We might actually need to find a way to transfer the tokens into the join address anyway 🙁 (but not in this PR already)